### PR TITLE
Refactor Encoding

### PR DESCRIPTION
**Pull request by tekand**
Refactor Paradox Encodings.
Encodings available in this PR:
- `paradox-en`
- `paradox-hu`
- `paradox-pt`
- `paradox-pl`
- `paradox-ee`
- `paradox-tr`
- `paradox-de`

special chars:
image

image

Disclaimer:
Some might lack of completeness (for example Í is missing from the Hungarian). The reason is that Paradox's map is also not complete.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@
 require-python-3
 construct~=2.9.43
 argparse>=1.4.0
+python-slugify>=4.0.1
```
Comment (yozik04) [inline requirements.txt:7]: This needs to be in setup.py as well.

File: tests/lib/test_encodings.py [hunk 2]
```diff
@@ -37,26 +39,26 @@ def test_en_encoding(raw, expected):
 
 
 def test_en_encoding_len():
-    assert len(en.decoding_table) == 224
-
+    assert len(en.charmap) == 256
 
-test_data_ru = [
```
Comment (yozik04) [inline tests/lib/test_encodings.py:43]: Anything wrong with Russian encoding?

**Review by yozik04 — COMMENTED** [review]
I am 95% agree. Looks perfect. Some notes see comments.

File: tests/lib/test_encodings.py (hunk 2 above)
Comment (tekand) [reply tests/lib/test_encodings.py:43]: No nothing, I commented out to make sure that the tests are not failing. Once I make the RU mapping, I'll uncomment it.

**Comment by yozik04** [comment]
Merged! Thank you!

**Pull request merged**